apps/cowswap-frontend/src/modules/permit/hooks/useCheckHasValidPendingPermit.ts

```diff
@@ -35,7 +48,7 @@ export function useCheckHasValidPendingPermit(): CheckHasValidPendingPermit {
 
       return checkHasValidPendingPermit(order, provider, chainId, permitInfo)
```

Comment:
Actually, you can pass here: `preGeneratedPermitInfo || permitInfo`

And on line `44` change it from `if (permitInfo === undefined)` to `if (permitInfo === undefined && preGeneratedPermitInfo === undefined)`

And on line 71 change from `!permitInfo` to `!isSupportedPermitInfo(permitInfo)`

And remove the `if` from line `40`